x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptFieldMapperTests.java

```diff
@@ -93,27 +95,43 @@ public void testStoredScriptsAreNotSupported() throws Exception {
         );
     }
 
-    public void testDefaultMapping() throws Exception {
-        XContentBuilder mapping = XContentFactory.jsonBuilder()
-            .startObject()
-            .startObject("_doc")
-            .startObject("properties")
-            .startObject("field")
-            .field("type", "script")
-            .field("runtime_type", randomFrom(SUPPORTED_RUNTIME_TYPES))
-            .startObject("script")
-            .field("source", "value('test')")
-            .field("lang", "test")
-            .endObject()
-            .endObject()
-            .endObject()
-            .endObject()
-            .endObject();
+    public void testKeyword() throws IOException {
+        MapperService mapperService = createIndex("test", Settings.EMPTY, mapping("keyword")).mapperService();
+        FieldMapper mapper = (FieldMapper) mapperService.documentMapper().mappers().getMapper("field");
+        assertThat(mapper, instanceOf(ScriptFieldMapper.class));
+        assertEquals(Strings.toString(mapping("keyword")), Strings.toString(mapperService.documentMapper()));
+    }
 
-        MapperService mapperService = createIndex("test", Settings.EMPTY, mapping).mapperService();
+    public void testLong() throws IOException {
+        MapperService mapperService = createIndex("test", Settings.EMPTY, mapping("long")).mapperService();
         FieldMapper mapper = (FieldMapper) mapperService.documentMapper().mappers().getMapper("field");
         assertThat(mapper, instanceOf(ScriptFieldMapper.class));
-        assertEquals(Strings.toString(mapping), Strings.toString(mapperService.documentMapper()));
+        assertEquals(Strings.toString(mapping("long")), Strings.toString(mapperService.documentMapper()));
+    }
+
+    private XContentBuilder mapping(String type) throws IOException {
+        XContentBuilder mapping = XContentFactory.jsonBuilder().startObject();
+        {
+            mapping.startObject("_doc");
+            {
+                mapping.startObject("properties");
+                {
+                    mapping.startObject("field");
+                    {
+                        mapping.field("type", "script").field("runtime_type", type);
+                        mapping.startObject("script");
+                        {
+                            mapping.field("source", "dummy_source").field("lang", "test");
+                        }
+                        mapping.endObject();
+                    }
+                    mapping.endObject();
+                }
+                mapping.endObject();
+            }
+            mapping.endObject();
+        }
+        return mapping.endObject();
```

Comment:
we are going to have ton of conflicts here with #59672, but I like the direction :)